Gauge/Source/RUsage.hsc

```diff
@@ -147,10 +148,8 @@ clongToW64 = fromIntegral
 sizeRUsage :: Int
 sizeRUsage = #const sizeof(struct rusage)
 
-pattern Self :: Who
```

Comment:
This one looks slightly dangerous. Is that necessary for 7.8, or is that the syntax is different on 7.8 so it's easier to remove the signature than doing a conditional signature ?

Reply:
I'm afraid I can't find a reference on the syntax change but with GHC-7.8 I get

```
/home/simon/src/hs-gauge/Gauge/Source/RUsage.hsc:151:14: parse error on input ‘::’
```

How about

```
pattern Self = (#const RUSAGE_SELF) :: Who
```

for a compromise? 

I can also include the signature conditionally with CPP.

Reply:
https://ghc.haskell.org/trac/ghc/ticket/8584 went in GHC-7.10